### PR TITLE
Migrate testing_codelab to Integration testing

### PR DESCRIPTION
**Pull request by abd99**
Added new integration tests based on the new package. Will remove the old tests when the update to the codelab goes live.

/cc @domesticmouse

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]
LGTM, but I'd love a review from @johnpryan to confirm =)

File: testing_codelab/step_07/integration_test/driver.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2021 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:integration_test/integration_test_driver.dart';
+
+Future<void> main() {
+  return integrationDriver(
+    responseDataCallback: (data) async {
```
Comment (johnpryan) [inline testing_codelab/step_07/integration_test/driver.dart:9]: This is more of a workaround for https://github.com/flutter/flutter/issues/74930, I would add a comment indicating that, or remove this altogether unless the codelab involves opening this in chrome://tracing.

File: testing_codelab/step_07/integration_test/app_test.dart [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    testWidgets('Scrolling test', (tester) async {
+      await tester.pumpWidget(TestingApp());
+
+      final listFinder = find.byType(ListView);
+      final scroller = tester.widget<ListView>(listFinder).controller;
+
+      await binding.traceAction(() async {
+        await binding.watchPerformance(() async {
+          await scroller.animateTo(
```
Comment (johnpryan) [inline testing_codelab/step_07/integration_test/app_test.dart:25]: I think we should recommend using [`tester.drag()`](https://api.flutter.dev/flutter/flutter_test/WidgetController/drag.html) instead and remove the `controller: ScrollController(),` lines from the app

File: testing_codelab/step_07/integration_test/app_test.dart [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          await tester.pumpAndSettle();
+        }, reportKey: 'scrolling_summary');
+      }, reportKey: 'scrolling_timeline');
+    });
+
+    testWidgets('Favorites operations test', (tester) async {
+      await tester.pumpWidget(TestingApp());
+
+      await binding.traceAction(() async {
+        await binding.watchPerformance(() async {
```
Comment (johnpryan) [inline testing_codelab/step_07/integration_test/app_test.dart:46]: I would try to show an example of 1. a test and 2. a performance benchmark. The first test is a benchmark, and this looks more like a test, so I would remove the `traceAction` and watchPerformance` wrappers.

**Review by johnpryan — CHANGES_REQUESTED** [review]

File: testing_codelab/step_07/integration_test/driver.dart (hunk 1 above)
Comment (abd99) [reply testing_codelab/step_07/integration_test/driver.dart:9]: I've mentioned this in the codelab note. I can still remove it if it feels unnecessary or something we should avoid for a codelab.

These are the exact word that I've used:
"Note: Writing a custom implementation for the responseDataCallback isn't necessary. However, doing so allows us to store the test data into separate files which makes it easier to further evaluate the test reports. If you omit this step, all the response data will be stored in one single file named integration_response_data.json."

File: testing_codelab/step_07/integration_test/app_test.dart (hunk 2 above)
Comment (abd99) [reply testing_codelab/step_07/integration_test/app_test.dart:25]: I actually used the pre-existing E2E test in the testing sample as an inspiration to write these new tests. How can I forget that there are easier ways! Thanks for suggesting this. I'm using `tester.fling()` now.

File: testing_codelab/step_07/integration_test/app_test.dart (hunk 3 above)
Comment (abd99) [reply testing_codelab/step_07/integration_test/app_test.dart:46]: This makes complete sense. I'll change it. Thanks!

**Comment by abd99** [comment]
PTAL @johnpryan, I've made the changes. I would also recommend you take a look at the codelab while reviewing this as it will help you understand how this code is being used. Brett has tagged you there.

**Review by johnpryan — APPROVED** [review]

**Comment by abd99** [comment]
PTAL @johnpryan, I have resolved the conflicts and made the same changes as the testing sample. The codelab doc is also in a much better place to be released now.

File: testing_codelab/step_05/pubspec.yaml [hunk 4]
```diff
@@ -23,6 +23,7 @@ dev_dependencies:
     sdk: flutter
   flutter_driver: # new
```
Comment (domesticmouse) [inline testing_codelab/step_05/pubspec.yaml:24]: Can we line up the `#new`s? Makes for easier cut'n'paste into the codelab doc =)

File: testing_codelab/step_04/pubspec.yaml [hunk 5]
```diff
@@ -23,6 +23,7 @@ dev_dependencies:
     sdk: flutter
   flutter_driver: # new
     sdk: flutter # new
+  integration_test: ^1.0.1    # new
```
Comment (domesticmouse) [inline testing_codelab/step_04/pubspec.yaml:26]: Also these `#new`s

File: testing_codelab/step_06/pubspec.yaml [hunk 6]
```diff
@@ -23,6 +23,7 @@ dev_dependencies:
     sdk: flutter
   flutter_driver: # new
     sdk: flutter # new
+  integration_test: ^1.0.1    # new
```
Comment (domesticmouse) [inline testing_codelab/step_06/pubspec.yaml:26]: And here

File: testing_codelab/step_07/pubspec.yaml [hunk 7]
```diff
@@ -23,6 +23,7 @@ dev_dependencies:
     sdk: flutter
   flutter_driver: # new
     sdk: flutter # new
+  integration_test: ^1.0.1    # new
```
Comment (domesticmouse) [inline testing_codelab/step_07/pubspec.yaml:26]: and here.

**Review by domesticmouse — APPROVED** [review]

File: testing_codelab/step_05/pubspec.yaml (hunk 4 above)
Comment (abd99) [reply testing_codelab/step_05/pubspec.yaml:24]: Of course, we can. Pushing changes soon.

**Review by johnpryan — APPROVED** [review]

**Comment by domesticmouse** [comment]
Landing. Thanks @abd99!

**Pull request merged**